src/hotspot/os/linux/compilerThreadTimeout_linux.cpp

```diff
@@ -40,15 +39,16 @@ void compiler_signal_handler(int signo, siginfo_t* info, void* context) {
 void CompilerThreadTimeoutLinux::compiler_signal_handler(int signo, siginfo_t* info, void* context) {
   switch (signo) {
     case TIMEOUT_SIGNAL: {
-      assert(false, "compile task timed out");
+      CompileTask* task = CompilerThread::current()->task();
+      assert(false, "compile task %d (%s) timed out after " INTPTR_FORMAT " ms",
+             task->compile_id(), task->method()->name_and_sig_as_C_string(), CompileTaskTimeout);
```

Comment:
Normally, you would probably need a `ResourceMark` for getting the method name. However, I'm not sure if you can do that as well here inside the signal handler. Maybe someone else can comment on that.